File: src/engine/commands/opencommand.cpp
#include "opencommand.h"


#define super Command

OpenCommand::OpenCommand(QObject *parent) :
    super(parent) {

    setDescription("Open an exit, typically a door. Note that doors automatically close after a "
                   "while.\n"
                   "\n"
                   "Example: open door");
}

OpenCommand::~OpenCommand() {
}

void OpenCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    GameObjectPtr exit = takeObject(currentRoom()->exits());
    if (!requireSome(exit, "Open what?")) {
        return;
    }

    player->open(exit);
}

File: data/commands/opencommand.js
function OpenCommand() {
    Command.call(this);

    this.setDescription("Open an exit, typically a door or a window. Note that doors " +
                        "automatically close after a while.\n" +
                        "\n" +
                        "Example: open door");
}

OpenCommand.prototype = new Command();
OpenCommand.prototype.constructor = OpenCommand;

OpenCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    var portal = this.takeObject(this.currentRoom.portals);
    if (!this.requireSome(portal, "Open what?")) {
        return;
    }

    player.open(portal);
};

CommandRegistry.registerCommand("open", new OpenCommand());
